lib/doorkeeper/oauth/client_authentication/client_secret_basic.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module OAuth
+    module ClientAuthentication
+      class ClientSecretBasic
+        def self.matches_request?(request)
+          request.authorization.present? && request.authorization.downcase.start_with?('basic')
+        end
+
+        def authenticate(request)
+          value = request.authorization.to_s.split(" ", 2).second
+          client_id, client_secret = Base64.decode64(value).split(':', 2)
+
+          return unless client_id.present? && client_secret.present?
```

Comment:
I'm not 100% sure if you can do client authentication with just the username, e.g.,

```
Authorization: Basic Base64Encode(pPqlv1hayMY3NXSMnikiqMLw3G2tyibuRex2HHo3aVE:)
```

My reading of the spec is that without the password this mechanism wouldn't match? Though, previously the code would have accepted that as a value authentication, and it'd return credentials without the `client_secret`, so maybe just the `client_id` (username) is necessary?

Reply:
I've asked about this here: https://mailarchive.ietf.org/arch/msg/oauth/QGJfkCMqN2kVMRMDwvCiF3mZ6r4/

Reply:
Consensus in the mailing list seems to imply that these methods only support POST requests, and require the `client_secret` to be present.